apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -1022,21 +1022,18 @@ private static MutateRowsRequest CreateMutateRowRequest(TableName tableName, IEn
             CallSettings callSettings,
             BigtableServiceApiClient.MutateRowsStream result)
         {
-            var defaultCallSettings = (UnderlyingClientSettings ?? new BigtableServiceApiSettings()).MutateRowSettings;
-            var effectiveCallSettings = defaultCallSettings.MergedWith(callSettings);
-            var retrySettings = effectiveCallSettings?.Timing?.Retry;
-            if (retrySettings != null)
-            {
-                effectiveCallSettings = effectiveCallSettings.WithCallTiming(CallTiming.FromExpiration(retrySettings.TotalExpiration));
-            }
+            var defaultSettings = UnderlyingClientSettings ?? new BigtableServiceApiSettings();
+            var effectiveCallSettings = defaultSettings.MutateRowsSettings.MergedWith(callSettings);
+            // TODO(mdour): Do we want to try to support per-call retry settings?
+            var effectiveRetrySettings = defaultSettings.MutateRowsRetrySettings;
 
             // TODO: Should the request manager be using the retry filter? We would need to fabricate RpcExceptions
             StatusCode[] retryStatuses = { StatusCode.DeadlineExceeded, StatusCode.Unavailable };
             var requestManager = new BigtableMutateRowsRequestManager(retryStatuses, request);
 
             var currentStream = result;
             var status = await ProcessCurrentStream().ConfigureAwait(false);
-            if (retrySettings != null && status == ProcessingStatus.Retryable)
+            if (effectiveRetrySettings != null && status == ProcessingStatus.Retryable)
```

Comment:
Check that the settings are actually retrying ones? Or does RetryOperationUntilCompleted handle single deadline settings anyway?

Reply:
These are `RetrySettings`, not `CallSettings`. It is possible to have a `RetrySettings` instance that doesn't retry?

Reply:
Whoops, hadn't spotted that - sorry.